modules/renter/proto/contract.go

```diff
@@ -345,8 +337,9 @@ func (cs *ContractSet) managedInsertContract(h contractHeader, roots []crypto.Ha
 		return modules.RenterContract{}, err
 	}
 	// write roots
-	for i, root := range roots {
-		if _, err := f.WriteAt(root[:], contractHeaderSize+crypto.HashSize*int64(i)); err != nil {
+	merkleRoots := newMerkleRoots(f)
+	for _, root := range roots {
+		if err := merkleRoots.push(root); err != nil {
```

Comment:
need to add a TODO somewhere in this package that currently the startup routine involves reading all of the merkle roots into memory, then hashing all of the merkle roots in the the `merkleRoots` object. That's a slow process that we should fix at some point, but it's probably not necessary until after we fix up the file structures.